angularjs-portal-frame/src/main/webapp/portal/main/controllers.js

```diff
@@ -61,6 +62,19 @@ define(['angular'], function(angular) {
       that.user = result;
     });
   }]);
+  
+  app.controller('WelcomeController', ['$localStorage', '$sessionStorage','$scope', '$document', 'APP_FLAGS', '$modal', function($localStorage, $sessionStorage, $scope, $document, APP_FLAGS, $modal) {
+    $scope.openModal = function() {
+      if (APP_FLAGS.welcome && !$localStorage.hasSeenWelcome) {
+        $modal.open({
+          animation: $scope.animationsEnabled,
+          templateUrl: require.toUrl('./partials/welcome.html'),
+          size: 'lg',
+        });
+        $localStorage.hasSeenWelcome = true;
```

Comment:
Probably an unimportant nitpick, but: would it be better to flip this bit on `$dismiss()` rather than on render, so that a user who hasn't dismissed this message sees it again if they reload the page, until they dismiss it?